File: SeleniumPomCucumberJunit5Allure/src/main/java/demoqa/pages/ElementsPage.java
package demoqa.pages;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;
import org.openqa.selenium.support.PageFactory;

public class ElementsPage extends BasePage {

	@FindBy(xpath = "//span[text()='Text Box']")
	private WebElement textBoxLink;
	
	@FindBy(xpath = "//span[text()='Web Tables']")
	private WebElement webTablesLink;

	public ElementsPage(WebDriver webDriver) {
		super(webDriver);
		PageFactory.initElements(webDriver, this);
	}

	public void goToTextBoxPage() {
		webDriverWait.until(element -> {
			return textBoxLink.isDisplayed();
		});
		textBoxLink.click();
	}
	
	public void goToWebTablesPage() {
		webDriverWait.until(element -> {
			return webTablesLink.isDisplayed();
		});
		webTablesLink.click();
	}

}

File: DotNetSeleniumC#Nunit/DotNetSeleniumC#Nunit/Pages/ElementsPage.cs
using OpenQA.Selenium;

namespace DotNetSelenium.pages
{
    public class ElementsPage
    {
        private readonly IWebDriver _driver;

        public ElementsPage(IWebDriver driver)
        {
            this._driver = driver;
        }

        private IWebElement TextBoxBtn => _driver.FindElement(By.XPath("//span[text()='Text Box']"));
        private IWebElement WebTablesBtn =>  _driver.FindElement(By.XPath("//span[text()='Web Tables']"));

        public void GoToTexBoxForm()
        {
            TextBoxBtn.Click();
        }

        public void GoToWebTablesPage()
        {
            WebTablesBtn.Click();
        }

    }
}
